### ISSUE TITLE
Feature: storing user's and media's watch providers

### ISSUE THREAD

**Issue by phirestalker** [bonukai/MediaTracker#499]
I had planned to get started adding a feature to eventually allow the lists to be filtered by what streaming services they are available on. I ran into a little snag, and I wanted your opinion on how it should be dealt with.

In the readme it says that ISO 639-1 is used for all the language settings. The examples only show the language part and not the country code. I was wondering if you would rather change the tmdb language setting to use the full language-country (eg en-us), or would you rather add a configuration option that lets users set their country to filter the streaming providers?

I wish their was some community driven place to find the deep links, but all the providers keep that for paying customers.

I will work on setting up the changes needed retrieving and storing the new information in the meantime.

**Comment by BanCrash** [comment]
> I had planned to get started adding a feature to eventually allow the lists to be filtered by what streaming services they are available on. I ran into a little snag, and I wanted your opinion on how it should be dealt with.
> 
> In the readme it says that ISO 639-1 is used for all the language settings. The examples only show the language part and not the country code. I was wondering if you would rather change the tmdb language setting to use the full language-country (eg en-us), or would you rather add a configuration option that lets users set their country to filter the streaming providers?
> 
> I wish their was some community driven place to find the deep links, but all the providers keep that for paying customers.
> 
> I will work on setting up the changes needed retrieving and storing the new information in the meantime.

IMO the best way is to add a new setting because someone could want the data from TMDB in english but they are actually living in France for example, so even having the full TMDB code wouldn't let them know what services are on their country.

**Comment by bonukai** [comment]
I agree with BenCrash, there should be a separate setting for this.

**Comment by phirestalker** [comment]
Agreed. Also, to add configuration options, does that need any migrations, or is it all handled by the configuration system? Can you point me in the right direction? I will need to add a country selector, as well as a list of streaming provider ids.

For the logos I intend to add another type to the current image fetching system. From the TMDB docs it looks like all images (including logos) are handled the same. This way I will not need a separate table for them.

**Comment by phirestalker** [comment]
Great. Let me know when the changes are ready, and I will see if I can get started on this.

**Comment by bonukai** [comment]
@phirestalker it's done, see details in #511
